nengo/tests/test_learning_rules.py

```diff
@@ -466,3 +467,61 @@ def test_frozen():
 
     with pytest.raises((ValueError, RuntimeError)):
         a.learning_rate = 1e-1
+
+
+@pytest.mark.parametrize('modifies', ["decoders", "weights"])
+def test_generic_rule(Simulator, nl_nodirect, seed, modifies):
+    if nl_nodirect == nengo.RectifiedLinear and modifies == "weights":
+        # this is a sloppy way of avoiding this one case without having
+        # to make a big custom parameterization
+        # TODO: why doesn't it work for this case?
```

Comment:
I didn't work too hard on it.  My guess is that it's because you end up with much larger gains on `RectifiedLinear` neurons, and that (perhaps combined with the lack of saturation) makes things more unstable.